### PR TITLE
Replace goblin crate with object

### PR DESCRIPTION
**Pull request by lnicola**
Continuation of #4385.

### CODE REVIEW

File: crates/proc_macro_srv/Cargo.toml [hunk 1]
```diff
@@ -10,9 +10,9 @@ edition = "2018"
 doctest = false
 
 [dependencies]
-goblin = "0.2.1"
 libloading = "0.6.0"
 memmap = "0.7"
+object = { version = "0.22", git = "https://github.com/gimli-rs/object/" }
```
Comment (bjorn3) [inline crates/proc_macro_srv/Cargo.toml:15]: Why use a git dependency?

Comment (lnicola) [reply crates/proc_macro_srv/Cargo.toml:15]: To pull in https://github.com/gimli-rs/object/pull/261 for https://github.com/gimli-rs/object/issues/222.

**Comment by matklad** [comment]
bors try

**Comment by lnicola** [comment]
Hmm, we can merge it like this or wait until a new `object` is released.

**Comment by matklad** [comment]
Why the current (0.23) is not good enough?

**Comment by lnicola** [comment]
It is, but with the fix in https://github.com/gimli-rs/object/pull/261 we could probably disable the `unaligned` feature.

**Comment by matklad** [comment]
I'd default to merging earlier and seding a followup, but you obv have more context here :)

**Comment by lnicola** [comment]
bors r=matklad

**Comment by philipc** [comment]
> It is, but with the fix in gimli-rs/object#261 we could probably disable the unaligned feature.

That fix made it into the 0.23 release.

**Pull request merged**